GitCommands/Git/GitGpgController.cs

```diff
@@ -0,0 +1,213 @@
+        {
+            _module = module;
+            _revision = revision;
+
+            TagVerifyMessage = "";
+        }
+
+        /// <summary>
+        /// Obtain the commit signature status on current revision.
+        /// </summary>
+        /// <returns>Enum value that indicate the gpg status for current git revision.</returns>
+        public CommitStatus GetRevisionCommitSignatureStatus()
+        {
+            CommitStatus cmtStatus = CommitStatus.NoSignature;
+            string gpg = _module.RunGitCmd($"log --pretty=\"format:%G?\" -1 {_revision.Guid}");
+
+            switch (gpg)
+            {
+                case GOOD_SIGN:         // "G" for a good (valid) signature
+                    cmtStatus = CommitStatus.GoodSignature;
+                    break;
+                case BAD_SIGN:          // "B" for a bad signature
+                case UNK_SIGN_VALIDITY: // "U" for a good signature with unknown validity 
+                case EXPIRED_SIGN:      // "X" for a good signature that has expired
+                case EXPIRED_SIGN_KEY:  // "Y" for a good signature made by an expired key
+                case REVOKED_KEY:       // "R" for a good signature made by a revoked key
+                    cmtStatus = CommitStatus.SignatureError;
+                    break;
+                case MISSING_PUB_KEY:   // "E" if the signature cannot be checked (e.g.missing key)
+                    cmtStatus = CommitStatus.MissingPublicKey;
+                    break;
+                case NO_SIGN:           // "N" for no signature
+                default:
+                    cmtStatus = CommitStatus.NoSignature;
+                    break;
+            }
+
+            return cmtStatus;
+        }
+
+
+        /// <summary>
+        /// Obtain the tag status on current revision.
+        /// </summary>
+        /// <returns>Enum value that indicate if current git revision has one tag with good signature, one tag with bad signature or more than one tag.</returns>
+        public TagStatus GetRevisionTagSignatureStatus()
+        {
+            TagStatus tagStatus = TagStatus.NoTag;
+
+            List<IGitRef> usefulRef = _revision.Refs.Where(x => x.IsTag && x.IsDereference).ToList<IGitRef>();
+
+            /* No Tag present, exit */
+            if (usefulRef.Count == 0)
+            {
+                return tagStatus;
+            }
+
+            Match goodSignatureMatch = null;
+            Match validSignatureMatch = null;
+
+            /* More than one tag on the revision */
+            if (usefulRef.Count > 1)
+            {
+                tagStatus = TagStatus.Many;
+
+                /* Only to populate TagVerifyMessage */
+                foreach (var gitRef in usefulRef)
+                {
+                    /* String printed in dialog box */
+                    TagVerifyMessage = $"{TagVerifyMessage}{gitRef.LocalName}\r\n{GetTagVerificationMessage(gitRef.LocalName, false)}\r\n\r\n";
```

Comment:
TagVerifyMessage is shown only on demand so it should be loaded lazily only on demand. 
`usefulRef` - should be then created in the constructor and kept as a private field to be reused.

Reply:
I'm having issue on lazy load and changing its value into cycle.

Reply:
I cannot have the same crash, this setting seems do nothing, revision detail is always below history graph.

Reply:
GitExt has to be restarted after changing this setting.

Reply:
do you mean completely exiting the debug session in VisualStudio and start it again ?

Reply:
Yes

Reply:
>I'm having issue on lazy load and changing its value into cycle.

It can be done like this: https://github.com/gpongelli/gitextensions/pull/4